### ISSUE TITLE
GitHub Continuous Integration - How Does it Work?

### ISSUE THREAD

**Issue by valkyrienyanko** [Valks-Games/io-game#42]
I want to auto deploy my app on every commit to the cloud on a port where everyone can connect to it and test it out.

I would also like to have an app to remind users to remove files like `package-lock.json` and `yarn.lock` from their pull requests.

I see many apps here, lots of them to explore, I don't know which to try out, and use to my advantage.

https://github.com/marketplace/category/continuous-integration

**Comment by aeuu** [comment]
One thing... You probably want to commit npm and yarn lock files, as they ensure that the resolved dependencies are consistent across different devices. Commit to one package manager and keep its lockfile in the repository.

**Comment by JohnTendik** [comment]
Yep @arctfoxx is right. You'll want to commit those files other wise everyone's environment is going to be different and you're going to run into issues where it works on one PC but doesn't on another. 

As for CI integration, I found Travis CI to be super straight forward and easy to integrate. They have a free tier option for open source projects which is a good deal :D
